### ISSUE TITLE
`Dynamic` cannot be used as a submodule name

### ISSUE THREAD

**Issue by scolsen** [carp-lang/Carp#865]
Defining a submodule named `Dynamic` results in a compiler error.

Given the source:
```
(defmodule Q                                                                                                                                                                                             
  (defmodule Dynamic                                                                                                                                                                                     
    (defn a [] "a")                                                                                                                                                                                      
  )                                                                                                                                                                                                      
)  
```

The carp compiler throws an error when attempting to evaluate the Dynamic submodule:

```
鲤 (load "d.carp")
鲤 :i Q
carp: Can't insert into non-existing module: Dynamic
CallStack (from HasCallStack):
  error, called at src/Lookup.hs:190:16 in CarpHask-0.3.0.0-JeTRwcTheISKNh9w0jc5KF:Lookup
```

Renaming the submodule solves the issue:

```
鲤 :i Q
Q : Module = {
    D : Module = {
        a : (Fn [] (Ref String a))
    }
}
Defined at line 0, column 0 in 'dummy-file'

```

**Comment by scolsen** [comment]
Note that it may be used as a top-level module name without issue.

**Comment by eriksvedang** [comment]
Hmm, may be something with the magic lookup in the Dynamic module.

**Comment by scolsen** [comment]
fixed in #1078
